adserver/views.py

```diff
@@ -2421,11 +2443,15 @@ def get_queryset(self):
         self.publisher = get_object_or_404(
             Publisher, slug=self.kwargs.get("publisher_slug", "")
         )
-        return self.publisher.user_set.all()
+        return (
+            UserPublisherMember.objects.filter(publisher=self.publisher)
+            .select_related()
+            .all()
```

Comment:
Are we doing this so we don't have to query the relationship again? Feels a bit fiddly than just passing the users in and querying the role from something on the user? I don't feel too strongly here, but just feels a bit complex.

Reply:
This is the authorized users view (a list view). It needs a list or queryset in order to for the class-based-view to do its thing. This is that queryset. It can't just be a list of users on the publisher anymore as it needs the role.